api.php

```diff
@@ -368,3 +386,51 @@ function resolveTags($tagscached)
 
 	return $tags;
 }
+
+function listOutOfDateMods($modsWithVersions)
+{
+	$outOfDateMods = array();
+	foreach($modsWithVersions as $modid => $version) {
+		$latestRelease = getLatestRelease($modid);
+		if (!empty($latestRelease) && $latestRelease["modversion"] > $version) {
+			$outOfDateMods[$modid] = $latestRelease;
+		}
+	}
+	good(array("statuscode" => 200, "updates" => $outOfDateMods));
+}
+
+function getLatestRelease($modid) {
+	global $con;
+
+	$latestRelease = $con->getRow("
+		select
+			`release`.*,
+			asset.*
+		from 
+			`release` 
+			join asset on (asset.assetid = `release`.assetid)
+		where modid=?
+		order by release.created desc
+	", array($modid));

```

Comment:
please make only one query over all modids, please e.g. with

```
[...]
where modid in (".join(",", array_fill(0, count($modids), "?")).")
order by release.created desc
", $modids);
```

Reply:
otherwise, nice! 👍

Reply:
All set! Changed so it queries once for all mod IDs :)